testCases/casual/casual/performance/test/casual.py

```diff
@@ -9,118 +9,126 @@
-      )
-      service_reply = requests.post(
-         url = host + "/.casual/service/state",
-         headers = { "content-type": "application/json"}
-      )
-      transaction_reply = requests.post(
-         url = host + "/.casual/transaction/state",
-         headers = { "content-type": "application/json"}
-      )
-
-      file_prefix = csv_prefix + "_" + domain['name']
-      # Domain metrics
-      with open( file_prefix + "_domain.metrics.json", "w") as file:
-         file.write( domain_reply.text)
-
-      # Service metrics
-      with open( file_prefix + "_service.metrics.json", "w") as file:
-         file.write( service_reply.text)
-
-      # Transaction metrics
-      with open( file_prefix + "_transaction.metrics.json", "w") as file:
-         file.write( transaction_reply.text)
-
-      if 'remote' in domain and 'type' in domain['remote'] and domain['remote']['type'] == "casual":
-        if not environment.parsed_options.use_remote_nodes:
-          helpers.execute( f"cp {domain['home']}/logs/statistics.log {csv_prefix}_{domain['name']}_statistics.log")
+    host = environment.host
+    csv_prefix = environment.parsed_options.csv_prefix
 
-        else:
-          source = domain['remote']['user'] + "@" + domain['remote']['host'] + ":" + domain['home']
+    for domain in configuration.get("domains", {}):
+        if 'casual_statistics' in domain and domain['casual_statistics'] == False:
+            continue
+
+        domain_reply = requests.post(
+            url = host + "/.casual/domain/state",
+            headers = { "content-type": "application/json"}
+        )
+        service_reply = requests.post(
+            url = host + "/.casual/service/state",
+            headers = { "content-type": "application/json"}
+        )
+        transaction_reply = requests.post(
+            url = host + "/.casual/transaction/state",
+            headers = { "content-type": "application/json"}
+        )
 
-          if source:
-            helpers.scp( f"{source}/logs/statistics.log", f"{csv_prefix}_{domain['name']}_statistics.log")
+        file_prefix = csv_prefix + "_" + domain['name']
+        # Domain metrics
+        with open( file_prefix + "_domain.metrics.json", "w") as file:
+            file.write( domain_reply.text)
 
+        # Service metrics
+        with open( file_prefix + "_service.metrics.json", "w") as file:
+            file.write( service_reply.text)
+
+        # Transaction metrics
+        with open( file_prefix + "_transaction.metrics.json", "w") as file:
+            file.write( transaction_reply.text)
+
+        if domain['lookup']['domain']['type'] == "casual":
+            if not environment.parsed_options.use_remote_nodes:
+                helpers.execute( f"cp {domain['home']}/logs/statistics.log {csv_prefix}_{domain['name']}_statistics.log")
+
+            else:
+                if source := domain['lookup']['host']['user'] + "@" + domain['lookup']['host']['hostname'] + ":" + domain['home']:
```

Comment:
I started to use this := operator to be able to both check and assign in one operation.
This is new in I think python 3.8, so we need to upgrade version on FK. It is available there but not installed.